include/astshim/Frame.h

```diff
@@ -1125,7 +1125,7 @@ class Frame : public Mapping {
     axis order for output data).
     */
     void permAxes(std::vector<int> perm) {
-        detail::assertEqual(static_cast<int>(perm.size()), "perm.size()", getNaxes(), "naxes");
+        detail::assertEqual(perm.size(), "perm.size()", static_cast<std::size_t>(getNaxes()), "naxes");
```

Comment:
Because AST itself uses an int (it has astGetI but not an unsigned variant). I consider unsigned ints a bit of a nuisance -- important if very large positive values are expected, but otherwise rarely worth the hassle. Note that much of the DM stack works the same way; for example Box.getArea, getWidth and getHeight all return int, despite the fact that they are never negative.